src/Datadog.Trace/Span.cs

```diff
@@ -405,11 +425,45 @@ internal Span SetMetric(string key, double? value)
                         Metrics = new Dictionary<string, double>();
                     }
 
-                    Metrics[key] = value.Value;
+                    if (value > -Pow2e53 && value < Pow2e53)
```

Comment:
Can you help me understand why we have this check? I thought a double could represent a wider range of values

Reply:
yeah the double could handle wider range of values, but is a limitation in golang side, for the metrics is being use float64 type with a limitation for precision, so we have to limit our data type to values that the go side can handle. There are a sample for that here: https://play.golang.org/p/fc5lmEOWu4n . The jira has a docs regarding this.

Reply:
Makes sense, thanks for explaining!